code/game/machinery/_machinery.dm

```diff
@@ -815,31 +814,40 @@
 	deconstruct(TRUE)
 
 /obj/machinery/deconstruct(disassembled = TRUE)
+	SHOULD_NOT_OVERRIDE(TRUE)
+
 	if(obj_flags & NO_DECONSTRUCTION)
+		dump_contents() //drop everything inside us
 		return ..() //Just delete us, no need to call anything else.
 
-	on_deconstruction()
+	on_deconstruction(disassembled)
 	if(!LAZYLEN(component_parts))
+		dump_contents() //drop everything inside us
 		return ..() //we don't have any parts.
 	spawn_frame(disassembled)
 
-	for(var/part in component_parts)
-		if(istype(part, /datum/stock_part))
-			var/datum/stock_part/datum_part = part
-			new datum_part.physical_object_type(loc)
-		else
-			var/obj/item/obj_part = part
-			component_parts -= part
-			obj_part.forceMove(loc)
-			if(istype(obj_part, /obj/item/circuitboard/machine))
-				var/obj/item/circuitboard/machine/board = obj_part
-				for(var/component in board.req_components) //loop through all stack components and spawn them
-					if(!ispath(component, /obj/item/stack))
-						continue
-					var/obj/item/stack/stack_path = component
-					new stack_path(loc, board.req_components[component])
+	if(length(component_parts))
```

Comment:
You don't need this `length(component_parts)`, `for(var/part in null)` or `for(var/part in list())` both do nothing anyways

Reply:
Yeah true. It felt weird but since it's a no op I'll just put it back then